zmon_worker_extras/check_plugins/exasol.py

```diff
@@ -4,89 +4,81 @@
 Query Exasol
 """
 
-import tempfile
-import subprocess
-import os
-
 from zmon_worker_monitor.adapters.ifunctionfactory_plugin import IFunctionFactoryPlugin, propartial
+from turbodbc import connect
 
 
 class ExaplusFactory(IFunctionFactoryPlugin):
     def __init__(self):
         super(ExaplusFactory, self).__init__()
         # fields from config
-        self._exacrm_cluster = None
-        self._exacrm_user = None
-        self._exacrm_pass = None
+        self._exasol_user = None
+        self._exasol_pass = None
+        self._exasol_driver = None
 
     def configure(self, conf):
         """
         Called after plugin is loaded to pass the [configuration] section in their plugin info file
         :param conf: configuration dictionary
         """
-        self._exacrm_cluster = conf['exacrm_cluster']
-        self._exacrm_user = conf['exacrm_user']
-        self._exacrm_pass = conf['exacrm_pass']
+        self._exasol_user = conf['exasol_user']
+        self._exasol_pass = conf['exasol_pass']
+        self._exasol_driver = conf['exasol_driver']
 
     def create(self, factory_ctx):
         """
         Automatically called to create the check function's object
         :param factory_ctx: (dict) names available for Function instantiation
         :return: an object that implements a check function
         """
-        return propartial(ExaplusWrapper, cluster=self._exacrm_cluster, password=self._exacrm_pass,
-                          user=self._exacrm_user)
+        return propartial(ExaplusWrapper, cluster=factory_ctx['cluster'], password=self._exasol_pass,
+                          user=self._exasol_user, schema=factory_ctx['schema'],
+                          driver=self._exasol_driver)
 
 
 class ExaplusWrapper(object):
-    def __init__(self, cluster, user='ZALANDO_NAGIOS', password='', schema=False):
+    def __init__(self, cluster, user='', password='', schema='', driver=''):
```

Comment:
``self._exasol_driver`` is not an attribute of the wrapper, only the factory. So it is not available in ``ExaplusWrapper``